lib/ykpiv.c

```diff
@@ -1567,7 +1567,11 @@ static ykpiv_rc _ykpiv_verify(ykpiv_state *state, char *pin, size_t *p_pin_len,
     return YKPIV_SIZE_ERROR;
   }
 
-  if (bio && verify_spin && (!pin || !p_pin_len || *p_pin_len != 16)) {
+  if(!bio && !pin) {
+    return YKPIV_AUTHENTICATION_ERROR;
+  }
+
+  if (pin && (!p_pin_len || *p_pin_len != 16)) {
```

Comment:
I'm not sure this is correct either.  This check fails is pin is not null and p_pin_len != 16, which should only be checked in the fingerprint session pin (bio && verify_spin) case.